src/js/graph/graph.js

```diff
@@ -95,93 +126,70 @@ Graph.prototype.removeNode = function(node) {
     node.removeGraph();
     this._graphImpl.removeNode(node.getId());
 };
-function logEdgeError(graph, sourceId, targetId, edgeType) {
-    var message;
-
-    if (!graph.getNodeById(sourceId) && !graph.getNodeById(targetId)) {
-        message = [
-            'Cannot create `',
-            edgeType,
-            '` edge between non-existent node ids `',
-            sourceId,
-            '` and `',
-            targetId,
-            '`.'
-        ].join('');
-    } else if (!graph.getNodeById(sourceId)) {
-        message = [
-            'Cannot create `',
-            edgeType,
-            '` edge from non-existent node id `',
-            sourceId,
-            '` to node id `',
-            targetId,
-            '`.'
-        ].join('');
-    } else if (!graph.getNodeById(targetId)) {
-        message = [
-            'Cannot create `',
-            edgeType,
-            '` edge from node id `',
-            sourceId,
-            '` to non-existent node id `',
-            targetId,
-            '`.'
-        ].join('');
+// Edge manipulation functions
+Graph.prototype.edgeExists = function(sourceId, targetId, edgeType) {
+    return this._graphImpl.hasEdge(sourceId, targetId, edgeType);
+};
+Graph.prototype.addEdge = function(sourceId, targetId, edgeType, toType) {
```

Comment:
It makes sense that we need to move edge logic out of nodes given the new underlying change. We should also move `Graph.addEdge` and `Graph.addEdges` to the semi-private namespace `Graph._addEdge` and `Graph._addEdges` to inform users of the library that they "should" not be used and instead should use the node-centric `Node.addEdge`.